spec/jobs/send_gpo_code_expiration_notices_job_spec.rb

```diff
@@ -0,0 +1,242 @@
+require 'rails_helper'
+
+RSpec.describe SendGpoCodeExpirationNoticesJob do
+  include Rails.application.routes.url_helpers
+
+  let(:job) { described_class.new(analytics: analytics) }
+
+  let(:analytics) { FakeAnalytics.new }
+
+  let(:usps_confirmation_max_days) { 30 }
+
+  let(:min_age_for_expiration_notice_in_days) { usps_confirmation_max_days.days + 1.day }
+
+  let(:expired_but_not_yet_notifiable_timestamp) { usps_confirmation_max_days.days.ago }
+
+  let(:expired_and_notifiable_timestamp) { min_age_for_expiration_notice_in_days.ago - 1.day }
+
+  let(:too_expired_to_notify_timestamp) { expired_but_not_yet_notifiable_timestamp - 3.days }
+
+  let(:not_expired_timestamp) { usps_confirmation_max_days.days.ago + 1.day }
+
+  let!(:user_with_expired_code_who_should_be_notified) do
+    create(:user, :with_pending_gpo_profile, code_sent_at: expired_and_notifiable_timestamp)
+  end
+
+  let!(:user_with_code_thats_not_expired_enough) do
+    create(:user, :with_pending_gpo_profile, code_sent_at: expired_but_not_yet_notifiable_timestamp)
+  end
+
+  let!(:user_with_two_expired_and_notifiable_codes) do
+    create(
+      :user,
+      :with_pending_gpo_profile,
+      code_sent_at: expired_and_notifiable_timestamp,
+    ).tap do |user|
+      profile = user.gpo_verification_pending_profile
+      create(
+        :gpo_confirmation_code,
+        profile: profile,
+        code_sent_at: expired_and_notifiable_timestamp - 1.second,
+      )
+    end
+  end
+
+  let!(:user_who_already_got_an_expiration_notice) do
+    create(
+      :user,
+      :with_pending_gpo_profile,
+      code_sent_at: expired_and_notifiable_timestamp,
+    ).tap do |user|
+      code = user.
+        gpo_verification_pending_profile.
+        gpo_confirmation_codes.first
+
+      code.update(expiration_notice_sent_at: Time.zone.now)
+    end
+  end
+
+  let!(:user_who_completed_gpo) do
+    create(
+      :user,
+      :with_pending_gpo_profile,
+      code_sent_at: expired_and_notifiable_timestamp,
+    ).tap do |user|
+      profile = user.gpo_verification_pending_profile
+      profile.remove_gpo_deactivation_reason
+      profile.activate
+    end
+  end
+
+  let(:deactivation_reasons) do
+    %i[
+      password_reset
+      encryption_error
+      verification_cancelled
+      gpo_verification_pending_NO_LONGER_USED
+      in_person_verification_pending_NO_LONGER_USED
+    ]
+  end
```

Comment:
Is there a way to use the Profile enum rather than duplicating it here?

Reply:
Addressed in 5d897244c932bbe6a2737ad7c6361937211de38e